imagetest/test_suites/image_validation/license_test.go

```diff
@@ -0,0 +1,200 @@
+	`Brocade Linux Fibre Channel HBA Firmware`,
+	`QLogic Linux Fibre Channel HBA Firmware`,
+	`Unlimited distribution and/or modification is allowed as long as this copyright notice remains intact.`,
+	`Permission is hereby granted to use.*this.*for any purpose`,
+	`are in the public domain`,
+	`is (available|distributed) under the terms of the GNU.*Public License`,
+	`(libudev|libgudev|udev) is licensed under the GNU (L|)GPL`,
+	`The Linux Console Tools are covered by the GPL`,
+	`Some portions of os-prober`,
+	`Netcat and the associated package is a product of Avian Research, and is freely available in full source form with no restrictions save an obligation to give credit where due.`,
+	`Permission is hereby granted, without written agreement and without licence or royalty fees, to use, copy, modify, and distribute this software`,
+	`Open Market permits you to use, copy, modify, distribute, and license this Software and the Documentation for any purpose, provided that existing copyright notices are retained in all copies and that this notice is included verbatim in any distributions. No written agreement, license, or royalty fee is required for any of the authorized uses.`,
+	`This software is made available under the terms of *either* of the licenses found in LICENSE.APACHE or LICENSE.BSD. Contributions to cryptography are made under the terms of *both* these licenses.`,
+}
+
+func isValidLicenseName(licenseCheck string) bool {
+	for _, name := range licenseNames {
+		var regexString = fmt.Sprintf("(?i)"+"(?:(?:License|Copyright)\\s*:\\s*{0})|(?:(?:covered )*under (?:the )?{0})|(?:under (?:the terms of )*the {%s})", name)
+		re := regexp.MustCompile(regexString)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidLicenseText(licenseCheck string) bool {
+	for _, licenseText := range licenses {
+		re := regexp.MustCompile(licenseText)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidLicense(licenseCheck string) bool {
+	return isValidLicenseName(licenseCheck) || isValidLicenseText(licenseCheck)
+}
+
+func TestArePackagesLicenseLegal(t *testing.T) {
+	if !isPackageLegal("/usr/share/doc/*/copyright") {
+		t.Fatalf("The packages are not legal to use")
+	}
+}
+
+func TestArePackagesCopyRightLegal(t *testing.T) {
+	if !isPackageLegal("/usr/share/doc/*/copyright") {
+		t.Fatalf("The packages are not legal to use")
+	}
+}
+
+func isPackageLegal(path string) bool {
+	var problemPackages []string
+	filenames, _ := filepath.Glob(path)
+	for _, filename := range filenames {
+		file, err := os.Open(filename)
+		if err != nil {
+			log.Printf("error open file")
+			continue
+		}
+		defer file.Close()
+
+		var licenseCheck string
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			licenseCheck += scanner.Text()
+		}
+
+		if err := scanner.Err(); err != nil {
+			log.Printf("error read file")
+			continue
+		}
+
+		re := regexp.MustCompile(`(\\*|#)*`)
```

Comment:
i don't think you need escapes in backtick-strings. and here you seem to be saying 0-or-more \ characters or a single \#, it doesn't seem to make sense as a regular expression. i encourage you not to try to create a 1-to-1 translation of the code you're basing this on but to understand what it does and implement it in go.

Reply:
the python regex you're copying from here is `r'(\*|#)*'`, that is a single asterisk or single \#

Reply:
```
Raw string literals are character sequences between back quotes, as in `foo`. Within the quotes,
any character may appear except back quote. The value of a raw string literal is the string
composed of the uninterpreted (implicitly UTF-8-encoded) characters between the quotes; in
particular, backslashes have no special meaning and the string may contain newlines. 
```
https://golang.org/ref/spec

Reply:
Good catch. I aware that it is trying to remove `*` or `#` with zero or more times. The double \`\ escapes `\` and change the semantics.